darwin/torch/dataset.py

```diff
@@ -182,8 +130,34 @@ def __str__(self):
         return format_string
 
 
+class BaseDataset(Dataset):
```

Comment:
If we don't fetch the dataset in classes' constructors anymore, then we wouldn't need this class